crates/red_knot_python_semantic/src/semantic_index.rs

```diff
@@ -187,6 +192,10 @@ impl<'db> SemanticIndex<'db> {
         self.symbol_tables[scope_id].clone()
     }
 
+    pub(super) fn instance_attribute_table(&self, scope_id: FileScopeId) -> Arc<SymbolTable> {
```

Comment:
The main motivation for using an `Arc` for the `SymbolTable` was so that we could return them from a salsa query that is specific per scope (instead of the entire `semantic_index`). However, it looks like no such query exists any more, so we should remove the `Arc` wrapper